contracts/PledgeAgent.sol

```diff
@@ -295,7 +298,7 @@ contract PledgeAgent is IPledgeAgent, System, IParamSubscriber {
   /// @param amount The amount of CORE to undelegate
   function undelegateCoin(address agent, uint256 amount) public {
     (uint256 deposit, ) = undelegateCoin(agent, msg.sender, amount, false);
-    Address.sendValue(payable(msg.sender), deposit);
```

Comment:
Avoiding potential reentrancy attack

Reply:
In the future testing PRs let's write some malicious contracts that attempt reentrancy exploits